python/ray/autoscaler/azure/node_provider.py

```diff
@@ -166,77 +177,44 @@ def create_node(self, node_config, tags, count):
         # TODO: restart deallocated nodes if possible
         location = self.provider_config["location"]
         resource_group = self.provider_config["resource_group"]
-        subnet_id = self.provider_config["subnet_id"]
 
-        config = node_config.copy()
-        config_tags = config.get("tags", {})
+        # load the template
+        template_path = os.path.join(
+            os.path.dirname(__file__), 'azure-vm-template.json')
+        with open(template_path, 'r') as template_file_fd:
+            template = json.load(template_file_fd)
+
+        # get the tags
+        config_tags = node_config.get("tags", {}).copy()
         config_tags.update(tags)
         config_tags[TAG_RAY_CLUSTER_NAME] = self.cluster_name
 
-        config["tags"] = config_tags
-        config["location"] = location
         name_tag = config_tags.get(TAG_RAY_NODE_NAME, "node")
-
-        for _ in range(count):
-            unique_id = uuid4().hex[:VM_NAME_UUID_LEN]
-            vm_name = "{name}-{id}".format(name=name_tag, id=unique_id)
-            config["os_profile"]["computer_name"] = vm_name
-
-            try:
-                assert len(vm_name) <= VM_NAME_MAX_LEN
-            except AssertionError as e:
-                e.args += ("name", vm_name)
-                raise
-
-            ip_configuration = {"name": uuid4(), "subnet": {"id": subnet_id}}
-
-            if not self.provider_config.get("use_internal_ips", False):
-                # create public ip address
-                public_ip_addess_params = {
-                    "location": location,
-                    "public_ip_allocation_method": "Dynamic"
-                }
-                public_ip_address = (
-                    self.network_client.public_ip_addresses.create_or_update(
-                        resource_group_name=resource_group,
-                        public_ip_address_name="{}-ip".format(vm_name),
-                        parameters=public_ip_addess_params).result())
-                ip_configuration["public_ip_address"] = public_ip_address
-
-            nic_params = {
-                "location": location,
-                "ip_configurations": [ip_configuration]
-            }
-            nic = self.network_client.network_interfaces.create_or_update(
-                resource_group_name=resource_group,
-                network_interface_name="{}-nic".format(vm_name),
-                parameters=nic_params).result()
-
-            # update vm config with network parameters
-            config["network_profile"] = {
-                "network_interfaces": [{
-                    "id": nic.id
-                }]
+        unique_id = uuid4().hex[:VM_NAME_UUID_LEN]
```

Comment:
is this just used to name the scale set? do the individual nodes inherit from this?